modules/home/konawall.nix

```diff
@@ -37,9 +49,14 @@ in with lib; {
           Requisite = PartOf;
         };
         Service = {
-          Environment = ["KONATAGS=${concatStringsSep "+" cfg.tags}"];
           Type = "oneshot";
-          ExecStart = "${cfg.package}/bin/konawall";
+          Environment = [
+            "PATH=${makeSearchPath "bin" (with pkgs; [ feh pkgs.xorg.xsetroot ])}"
+          ];
+          ExecStart = let
+            tags = map (n: concatStringsSep "," n) cfg.tagList;
+            tags-sep = concatStringsSep " " tags;
+          in "${cfg.package}/bin/konawall --mode ${cfg.mode} ${if ((length cfg.commonTags) > 0) then ''--common ${concatStringsSep "," cfg.commonTags}'' else ""} ${tags-sep}";
```

Comment:
`optionalString (cfg.commonTags != [ ]) "--common ${blah}"` is the typical way to do this

also pull the common tags out into the let too